flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/DataStatisticsOperator.java

```diff
@@ -139,14 +168,16 @@ public void handleOperatorEvent(OperatorEvent event) {
         operatorName,
         subtaskIndex,
         statisticsEvent.checkpointId());
-    globalStatistics =
+    this.globalStatistics =
         StatisticsUtil.deserializeAggregatedStatistics(
             statisticsEvent.statisticsBytes(), aggregatedStatisticsSerializer);
     checkStatisticsTypeMigration();
-    output.collect(new StreamRecord<>(StatisticsOrRecord.fromStatistics(globalStatistics)));
```

Comment:
it is a bug previously. we actually don't want to apply the new stats immediately during normal aggregation and propagation phase. switch happens at checkpoint boundary.

`applyImmediately` flag is added in this PR to distinguish the stats requested during rescale case. In this case, immediate application is desired.

Reply:
Why don't apply the new statistics immediately?

Reply:
Oh... i remember.  Don't want to mess with the ongoing files

Reply:
yeah. because Iceberg sink flush and commit at checkpoint boundary, switching at checkpoint boundary allows all subtasks to switch to the new stats for the same checkpoint cycle. Otherwise, some records are shuffled based on old stats and some are shuffled based on new stats.